onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -450,40 +450,56 @@ static Status PartitionOrtFormatModelImpl(Graph& graph, FuncManager& func_mgr,
 
   std::vector<NodeComputeInfo> node_compute_funcs;
   node_compute_funcs.reserve(nodes_and_viewers.size());
-
-  ORT_RETURN_IF_ERROR(current_ep.Compile(nodes_and_viewers, node_compute_funcs));
-
-  if (node_compute_funcs.size() != nodes_and_viewers.size()) {
-    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, type, " did not return correct number of compiled functions");
+  std::unordered_set<size_t> compile_error_indices;
+  compile_error_indices.reserve(nodes_and_viewers.size());
+
+  // We will compile the partitions one by one, and put the result back into node_compute_funcs
+  // In case one of the partitions fails the compile, we can fall back this particular partition to next EP
+  for (size_t i = 0; i < nodes_and_viewers.size(); ++i) {
```

Comment:
nit: 'partitions' is confusing given the set of nodes is defined by an IndexedSubGraph. 

  // We will compile the fused nodes one by one, and put the result back into node_compute_funcs if successful.
  // In a compilation fails we undo the fusion and leave the original nodes available for other EPs to take